server/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -61,7 +63,15 @@
 import static org.hamcrest.Matchers.nullValue;
 
 public class SearchHitTests extends AbstractStreamableTestCase<SearchHit> {
-    public static SearchHit createTestItem(boolean withOptionalInnerHits) {
+    /**
+     * Creates a random {@link SearchHit} that will preserve object equality
+     * upon rest (de)serialization if the provided {@link XContentType} is not null.
+     */
+    public static SearchHit createTestItem(@Nullable XContentType xContentType) {
+        return createTestItem(xContentType, false);
+    }
+
+    public static SearchHit createTestItem(@Nullable XContentType xContentType, boolean isInnerHit) {
```

Comment:
I get confused here about why we need to accept null xContentType, and why that converts to JSON. Could we randomize it all the time instead and have it non-nullable? Or provide JSON rather than null so we are explicit about it?